### PR TITLE
Add issues for extract and validate job

### PR DESCRIPTION
**Pull request by joelverhagen**
Complete https://github.com/NuGet/NuGetGallery/issues/5256.
Progress on https://github.com/NuGet/Engineering/issues/785.

This PR brings two main things:

1. Add `ValidatorIssues` table so that issues can be related to `ValidatorStatuses`.
1. Add two new issue codes for extract and validate:
   1. `SignedPackageMustHaveOneSignature`
   1. `ClientSigningVerificationFailure` - this essentially wraps a client error code and message.

<details>
<summary>New table SQL</summary>
<pre>
CREATE TABLE [dbo].[ValidatorIssues] (
    [Key] [bigint] NOT NULL IDENTITY,
    [ValidationId] [uniqueidentifier] NOT NULL,
    [IssueCode] [int] NOT NULL,
    [Data] [nvarchar](max) NOT NULL,
    CONSTRAINT [PK_dbo.ValidatorIssues] PRIMARY KEY ([Key])
)
CREATE INDEX [IX_ValidationId] ON [dbo].[ValidatorIssues]([ValidationId])
ALTER TABLE [dbo].[ValidatorIssues] ADD CONSTRAINT [FK_dbo.ValidatorIssues_dbo.ValidatorStatuses_ValidationId] FOREIGN KEY ([ValidationId]) REFERENCES [dbo].[ValidatorStatuses] ([ValidationId]) ON DELETE CASCADE
</pre>
</details>

### CODE REVIEW

File: src/NuGet.Services.Contracts/Validation/ValidationIssueCode.cs [hunk 1]
```diff
@@ -20,6 +20,16 @@ public enum ValidationIssueCode
         /// </summary>
         PackageIsSigned = 1,
 
+        /// <summary>
+        /// A signing verification failure occurred, where the issue message is mastered by the client APIs.
+        /// </summary>
+        ClientSigningVerificationFailure = 2,
```
Comment (skofman1) [inline src/NuGet.Services.Contracts/Validation/ValidationIssueCode.cs]: Why is it important for the consumer of the IssueCode (the Gallery) to know the error came from the client? Will a more general error code be sufficient?

File: src/NuGet.Services.Validation/Entities/ValidatorIssue.cs [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+namespace NuGet.Services.Validation
+{
+    public class ValidatorIssue : BaseValidationIssue
+    {
+        /// <summary>
+        /// The validation ID that this issue is related to. This is a foreign key to the <see cref="ValidatorStatus"/>
+        /// entity.
+        /// </summary>
+        public Guid ValidationId { get; set; }
```
Comment (skofman1) [inline src/NuGet.Services.Validation/Entities/ValidatorIssue.cs]: ValidationStatus contains ValidationId. Is this a duplication?

File: src/NuGet.Services.Contracts/Validation/ValidationIssueCode.cs (hunk 1 above)
Comment (joelverhagen) [reply src/NuGet.Services.Contracts/Validation/ValidationIssueCode.cs]: I don't feel strongly about the name, but in general issues should not have entire messages serialized in their data. The name is meant to remind that caller that this is an oddball that has limitations for (future) localization.

File: src/NuGet.Services.Validation.Issues/SignedPackageMustHaveOneSignature.cs [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+namespace NuGet.Services.Validation.Issues
+{
+    public class SignedPackageMustHaveOneSignature : ValidationIssue
+    {
+        [JsonConstructor]
+        public SignedPackageMustHaveOneSignature(int count)
+        {
+            if (count == 1)
+            {
+                throw new ArgumentOutOfRangeException(nameof(count));
```
Comment (skofman1) [inline src/NuGet.Services.Validation.Issues/SignedPackageMustHaveOneSignature.cs]: can you add a helpful message to the exception?

Comment (joelverhagen) [reply src/NuGet.Services.Validation.Issues/SignedPackageMustHaveOneSignature.cs]: Sure

Comment (joelverhagen) [reply src/NuGet.Services.Validation.Issues/SignedPackageMustHaveOneSignature.cs]: Fixed.

**Pull request merged**